### ISSUE TITLE
[Bug Report] ttnn untilize fails for Llama logits shape

### ISSUE THREAD

**Issue by cglagovichTT** [tenstorrent/tt-metal#11509]
**Describe the bug**
The following call fails due to L1 OOM error:
```
tt_logits = ttnn.to_layout(tt_logits, layout=ttnn.ROW_MAJOR_LAYOUT)
```
where `tt_logits: [32, 128k] bf16, dram interleaved`.  

This is because the untilize op parallelizes on tile rows of the input. It fails to allocate CBs to buffer the input and output of this op.

This issue is related to Llama e2e performance (#11386), as we would like to untilize on device to reduce e2e times.  

**To Reproduce**
1. branch `cglagovich/11509`
2. `pytest -svv tests/ttnn/unit_tests/test_to_layout.py`

**Expected behavior**
```
E       Statically allocated circular buffers on core range [(x=0,y=0) - (x=0,y=0)] grow to 33661792 B which is beyond max L1 size of 1499136 B
```

**Comment by cglagovichTT** [comment]
@eyonland could you please reassign to the op owner?

**Comment by eyonland** [comment]
@yan-zaretskiy mentioned that it’s related to the fact that the multicore untilize allocates CBs greedily (assuming it has enough room for the entire row of tiles). Single-core untilize does it conservatively, but this was not ported to the multicore version.

**Comment by cglagovichTT** [comment]
I'm raising this to P0 since it's limiting e2e performance which is relevant for t3k and TG demos

**Comment by uaydonat** [comment]
@tarafdarTT please prioritize or delagate if not possible

**Comment by uaydonat** [comment]
@tarafdarTT are you actively working on this? any eta?

**Comment by tarafdarTT** [comment]
I am actively working on this. ETA by end of week or sooner.

**Comment by uaydonat** [comment]
https://github.com/tenstorrent/tt-metal/pull/12123

Naif's comment:
FYI this should work, i think beecause of how the llk's work with blocks being in a row, its currently paralleleized in height still.
I am going to gather perf results and am working on a column parallelized version of it by tomorrow.
Lets sync tomorrow about this.

**Comment by uaydonat** [comment]
fyi @yieldthought
